From 1c6c152ea400ec870a359462e48edaebd6801353 Mon Sep 17 00:00:00 2001
From: "git@status.e4ward.com" <git@status.e4ward.com>
Date: Thu, 27 Sep 2012 19:29:45 -0400
Subject: [PATCH] UBUNTU: SAUCE: input: Cypress PS/2 Trackpad fix
 multi-source, double-click

Fix "Quickly clicking touchpad results in double click (with tap
to click)" by setting INPUT_PROP_BUTTONPAD property. This lets
xserver-xorg-input-synaptics know that the touchpad has a button
underneath it so it can take care of the multiple-click-source
issue.

BugLink: http://bugs.launchpad.net/bugs/1055788

Signed-off-by: Kyle Fazzari <git@status.e4ward.com>
Signed-off-by: Kamal Mostafa <kamal@canonical.com>
Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 drivers/input/mouse/cypress_ps2.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/input/mouse/cypress_ps2.c b/drivers/input/mouse/cypress_ps2.c
index 3a572ec..c3752be 100644
--- a/drivers/input/mouse/cypress_ps2.c
+++ b/drivers/input/mouse/cypress_ps2.c
@@ -423,6 +423,7 @@ static int cypress_set_input_params(struct input_dev *input,
 
 		}
 
+		__set_bit(INPUT_PROP_BUTTONPAD, input->propbit);
 		__set_bit(EV_KEY, input->evbit);
 		__set_bit(BTN_TOUCH, input->keybit);
 		__set_bit(BTN_TOOL_FINGER, input->keybit);
@@ -439,6 +440,7 @@ static int cypress_set_input_params(struct input_dev *input,
 		__clear_bit(REL_X, input->relbit);
 		__clear_bit(REL_Y, input->relbit);
 	} else {
+		__set_bit(INPUT_PROP_BUTTONPAD, input->propbit);
 		__set_bit(EV_REL, input->evbit);
 		__set_bit(REL_X, input->relbit);
 		__set_bit(REL_Y, input->relbit);
-- 
1.8.0

